src/Wallpaper/Portal.vala

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-FileCopyrigthText: 2024 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+[DBus (name = "org.freedesktop.impl.portal.Wallpaper")]
+public class Wallpaper.Portal : Object {
+    private DBusConnection connection;
+
+    public Portal (DBusConnection connection) {
+        this.connection = connection;
+    }
+
+    [DBus (name = "SetWallpaperURI")]
+    public async uint set_wallpaper_uri (
+        ObjectPath handle,
+        string app_id,
+        string parent_window,
+        string uri,
+        HashTable<string, Variant> options
+    ) throws DBusError, IOError {
+        unowned string set_on = "both"; // Possible values are background, lockscreen or both.
+        unowned bool show_preview = false; // FIXME: implement preview dialog
+
+        unowned var _set_on = options.get ("set-on");
+        if (_set_on.get_type_string () == "s") {
```

Comment:
I insist, but this means no string allocation ;)
```suggestion
        if (_set_on.is_of_type (GLib.VariantType.STRING)) {
```

Reply:
Oh that's way cleaner, thanks!